webapp/channels/src/components/size_aware_image.tsx

```diff
@@ -23,83 +26,103 @@ const MIN_IMAGE_SIZE = 48;
 const MIN_IMAGE_SIZE_FOR_INTERNAL_BUTTONS = 100;
 const MAX_IMAGE_HEIGHT = 350;
 
+export type Props = {
+
+    /*
+    * The source URL of the image
+    */
+    src: string;
+
+    /*
+    * dimensions object to create empty space required to prevent scroll pop
+    */
+    dimensions?: Partial<PostImage>;
```

Comment:
I am not sure if this should be of type PostImage because it doesnt contain other types defined in PostImage, This could be simple object type with width and heigts as keys?

Reply:
I just inspected wherever this component has been used and the type of dimension passed are:
![image](https://github.com/mattermost/mattermost/assets/93595710/6aabd51b-f711-478c-a9de-c9a6b0682e11)

<h2><h2/>

![image](https://github.com/mattermost/mattermost/assets/93595710/55299911-718e-4d09-bfd5-9e6389aaa244)